File: shared/src/main/java/ru/privatenull/pnauth/platform/PnPlatform.java
package ru.privatenull.pnauth.platform;

import ru.privatenull.pnauth.dialog.PlayerDialogs;
import java.util.Collection;
import java.util.Optional;
import java.util.UUID;

/** Entry point for player and platform services exposed to extensions. */
public interface PnPlatform {
    PlatformType type();
    Optional<PnPlayer> player(UUID uniqueId);
    Optional<PnPlayer> player(String username);
    Collection<PnPlayer> players();
    PlatformScheduler scheduler();
    TaskRegistry tasks();
    PlayerDialogs dialogs();
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/platform/PnPlatform.kt
package ru.privatenull.pnauth.platform

import ru.privatenull.pnauth.dialog.PlayerDialogs
import java.util.Optional
import java.util.UUID

/** Entry point for player and platform services exposed to extensions. */
interface PnPlatform {
    fun type(): PlatformType
    fun player(uniqueId: UUID): Optional<PnPlayer>
    fun player(username: String): Optional<PnPlayer>
    fun players(): Collection<PnPlayer>
    fun scheduler(): PlatformScheduler
    fun tasks(): TaskRegistry
    fun dialogs(): PlayerDialogs
}
